CMakeLists.txt

```diff
@@ -63,6 +64,10 @@ list (APPEND EKAT_TPL_LIBRARIES_INTERNAL ${YAMLCPP_LIBRARIES})
 BuildSpdlog()
 list (APPEND EKAT_TPL_LIBRARIES_INTERNAL ${SPDLOG_LIBRARIES})
 
+# EKAT offers Catch2 for unit testing
+BuildCatch2()
+list (APPEND EKAT_TPL_LIBRARIES_INTERNAL ${CATCH2_LIBRARIES})
```

Comment:
I was looking at this cmake var. It's used to then set the `EKAT_TPL_LIBRARIES` var, which is only used to link libs to ekat. I think that doing
```
targer_link_libraries (ekat PUBLIC
  ${Kokkos_LIBRARIES}
  ${YAMLCPP_LIBRARIES}
  ${SPDLOG_LIBRARIES}
  ${CATCH2_LIBRARIES}
)
```
might actually be clearer (and easier to maintain) than creating the two cmake vars (one of which is a cache one, for some reason without INTERNAL, so the user could modify it!)

Reply:
I tried to change this, but I think it messed up the dependency graph for the `ekat` target. Maybe we can fiddle with this in another cleanup PR, like the one in which we might separate out the Catch2 logic.